ui/src/app/lib/atlasmap-data-mapper/components/expression.component.ts

```diff
@@ -176,23 +176,7 @@ export class ExpressionComponent implements OnInit, OnDestroy, OnChanges {
     if (droppedField === null || currentFieldMapping === null || !droppedField.isSource) {
       return;
     }
-
-    if (droppedField.partOfMapping) {
-
-      // The selected field is part of a different mapping.
-      if (!currentFieldMapping.isFieldMapped(droppedField)) {
-        return;
-      }
-      // TODO handle drop position - for now this appends a field ref to the end of expression
-      const mappedField = currentFieldMapping.getMappedFieldForField(droppedField, true);
-      this.addConditionalExpressionNode(mappedField, null, 0);
-
-    // Pulling an unmapped field into a transition expression evaluation implies a compound selection.
-    } else {
-      // TODO handle drop position - for now this appends a field ref to the
-      // end of expression in FieldMappingPair#updateTransition()
-      this.configModel.mappingService.fieldSelected(droppedField, true);
-    }
+    this.configModel.mappingService.fieldSelected(droppedField, true);
```

Comment:
isn't this removing a field from mapping if it's already partOfMapping?

Reply:
Debugging through fieldSelected() - it will only remove the field of the selected source field is part of the active mapping which ar the point that its dropped into the conditional expression its not.

Reply:
but field reference could be added multiple times into same expression I think? like when writing an expression `IF(ISEMPTY(name), name, '')` and dropping a `name` field when filling 2nd argument?

Reply:
kinda related to #988

Reply:
Yup - that was the use case I was missing - trying again...